DotNet.DocsTools/GitHubClientServices/GitHubAppClient.cs

```diff
@@ -0,0 +1,69 @@
+﻿using DotNetDocs.Tools.GitHubCommunications;
+using GitHubJwt;
+using System.Net.Http.Headers;
+
+namespace DotNet.DocsTools.GitHubClientServices;
+
+internal class GitHubAppClient : GitHubClientBase, IGitHubClient, IDisposable
+{
+    private readonly int _appID;
+    private readonly string _oauthPrivateKey;
+
+    private Task regenerateTask = default!;
+    private CancellationTokenSource? _tokenSource;
+
+    public sealed class PlainStringPrivateKeySource : IPrivateKeySource
+    {
+        private readonly string _key;
+
+        public PlainStringPrivateKeySource(string key) => _key = key;
+
+        public TextReader GetPrivateKeyReader() => new StringReader(_key);
+    }
+
+    internal GitHubAppClient(int appID, string oauthPrivateKey) =>
+        (_appID, _oauthPrivateKey) = (appID, oauthPrivateKey);
+
+    internal async Task GenerateTokenAsync()
+    {
+        var privateKeySource = new PlainStringPrivateKeySource(_oauthPrivateKey);
+        var generator = new GitHubJwtFactory(
+            privateKeySource,
+            new GitHubJwtFactoryOptions
+            {
+                AppIntegrationId = _appID,
+                ExpirationSeconds = 8 * 60 // 600 is apparently too high
```

Comment:
So, we're able to request how long the token is alive before expiring? If so how does this hardcoded value correlate to the hardcoded five mins time span below?